rts/Lua/LuaOpenGL.cpp

```diff
@@ -5110,6 +5168,71 @@ int LuaOpenGL::ReadPixels(lua_State* L)
 }
 
 
+namespace Impl {
+	template<class Type>
+	static inline void ReadAttachmentPixel(lua_State* L, GLint x, GLint y, GLenum format, GLenum readType, int fSize) {
+		Type data[fSize];
+		glReadPixels(x, y, 1, 1, format, readType, data);
+		for (int e = 0; e < fSize; e++) {
+			lua_pushnumber(L, data[e]);
+		}
+	}
+}
+
+int LuaOpenGL::ReadAttachmentPixel(lua_State* L)
+{
+	GLenum attachment;
+	switch(hashString(luaL_checkstring(L, 1))) {
+		case hashString("color0"):  { attachment = GL_COLOR_ATTACHMENT0; } break;
+		case hashString("color1"):  { attachment = GL_COLOR_ATTACHMENT1; } break;
+		case hashString("color2"):  { attachment = GL_COLOR_ATTACHMENT2; } break;
+		case hashString("color3"):  { attachment = GL_COLOR_ATTACHMENT3; } break;
+		case hashString("color4"):  { attachment = GL_COLOR_ATTACHMENT4; } break;
+		case hashString("color5"):  { attachment = GL_COLOR_ATTACHMENT5; } break;
+		case hashString("color6"):  { attachment = GL_COLOR_ATTACHMENT6; } break;
+		case hashString("color7"):  { attachment = GL_COLOR_ATTACHMENT7; } break;
+		case hashString("color8"):  { attachment = GL_COLOR_ATTACHMENT8; } break;
+		case hashString("color9"):  { attachment = GL_COLOR_ATTACHMENT9; } break;
+		case hashString("color10"): { attachment = GL_COLOR_ATTACHMENT10; } break;
+		case hashString("color11"): { attachment = GL_COLOR_ATTACHMENT11; } break;
+		case hashString("color12"): { attachment = GL_COLOR_ATTACHMENT12; } break;
+		case hashString("color13"): { attachment = GL_COLOR_ATTACHMENT13; } break;
+		case hashString("color14"): { attachment = GL_COLOR_ATTACHMENT14; } break;
+		case hashString("color15"): { attachment = GL_COLOR_ATTACHMENT15; } break;
+		case hashString("depth"):   { attachment = GL_DEPTH_ATTACHMENT; } break;
+		default: assert(false);
+	}
+
+	const GLint x = luaL_checkint(L, 2);
+	const GLint y = luaL_checkint(L, 3);
+
+	const auto activeLuaFBO = CLuaHandle::GetActiveFBOs(L).GetActiveReadFBO();
+	assert(activeLuaFBO);
```

Comment:
Is some fallback FBO always active, or will this trigger if I call it inside e.g. GameFrame or other non-drawing context?